omnibus/package-scripts/agent/preinst

```diff
@@ -120,14 +120,52 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
     fi
 else
     # macOS
-    mkdir -p $LOG_DIR
+
+    # On Mac, the real log folder is located under the user profile folder, because some MacOS versions deletes
+    # /var/log on upgrade. The old log folder now becomes a symbolic link to the new folder.
+    REAL_LOG_DIR=/opt/datadog-agent/logs
+    LOG_BASE_DIR=/var/log
+
+    mkdir -p $REAL_LOG_DIR
 
     CONF_DIR="$INSTALL_DIR/etc"
     APP_DIR="/Applications/Datadog Agent.app"
 
-    LOG_FILE="$LOG_DIR/preinstall.log"
+    LOG_FILE="$REAL_LOG_DIR/preinstall.log"
     exec > $LOG_FILE 2>&1
 
+    if [ -d $LOG_DIR ]; then
```

Comment:
Is this code path used on both new installs and upgrades? It should, but I'm not sure now.

Reply:
It's used in both cases. But Mac doesn't track pkg file, unlike the rpm on Linux, so there's no check for upgrade.